components/pill/__docs__/storybook-stories.jsx

```diff
@@ -0,0 +1,316 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import IconSettings from '~/components/icon-settings';
+import { PILL } from '~/utilities/constants';
+import Pill from '~/components/pill';
+import Icon from '~/components/icon';
+import Avatar from '~/components/avatar';
+
+function noop () {
+}
+
+storiesOf(PILL, module)
+	.addDecorator((getStory) => <div className="slds-p-around--medium"><IconSettings iconPath="/assets/icons">{getStory()}</IconSettings></div>)
+	.add('Linked', () => (
+		<Pill
+			labels={{
+				label: 'Pill Label',
+				title: 'Full pill label verbiage mirrored here',
+				removeTitle: 'Remove'
+			}}
+			onClick={noop}
+			onRemove={noop}
+		/>
+	))
+	.add('Unlinked', () => (
+		<Pill
+			labels={{
+				label: 'Pill Label',
+				title: 'Full pill label verbiage mirrored here',
+				removeTitle: 'Remove'
+			}}
+			onRemove={noop}
+		/>
+	))
+	.add('With Icon', () => (
+		<Pill
+			labels={{
+				label: 'Pill Label',
+				title: 'Full pill label verbiage mirrored here',
+				removeTitle: 'Remove'
+			}}
+			icon={
+				<Icon
+					title="Account"
+					category="standard"
+					name="account"
+				/>
+			}
+			onClick={noop}
+			onRemove={noop}
+		/>
+	))
+	.add('With Avatar', () => (
+		<Pill
+			labels={{
+				label: 'Pill Label',
+				title: 'Full pill label verbiage mirrored here',
+				removeTitle: 'Remove'
+			}}
+			avatar={
+				<Avatar
+					variant="user"
+					title="User avatar"
+					imgSrc="https://lightningdesignsystem.com/assets/images/avatar2.jpg"
+				/>
+			}
+			onClick={noop}
+			onRemove={noop}
+		/>
+	))
+	.add('Error', () => (
+		<Pill
+			labels={{
+				label: 'Pill Label',
+				title: 'Full pill label verbiage mirrored here',
+				removeTitle: 'Remove'
+			}}
+			hasError
+			icon={
+				<Icon
+					title="Error"
+					category="utility"
+					name="warning"
+					className="slds-icon-text-error"
```

Comment:
In SLDS it only specifies a style for the pill's border and this is for the case you don't want the icon. So I left it out. It would make sense though if the pill similar to other components supported a notion of a variant  and there was the error variant.